src/main/java/org/jenkinsci/test/acceptance/po/PluginManager.java

```diff
@@ -285,7 +285,14 @@ private void tickPluginToInstall(PluginSpec spec) {
         WebElement filterBox = find(By.id("filter-box"));
         filterBox.clear();
         filterBox.sendKeys(name);
-        check(waitFor(by.xpath("//input[starts-with(@name,'plugin.%s.')]", name), 10));
+        WebElement pluginEntry = waitFor(by.xpath("//input[starts-with(@name,'plugin.%s.')]", name), 10);
+        // sometimes the web element gets stale because of the dynamic JS behaviour of the plugin manager UI
+        if (Control.isStale(pluginEntry)) {
+            // wait 1 second, re-try and give up if it fails again
+            sleep(1000);
```

Comment:
Either the comment is wrong (if it uses elastic time it may wait more or less than 1 second) or this is not using `elasticTime`and it should be :)
How often do we hit this - is waiting 1 second going to make the tests even slower?

Could this be better written using a `waitForCont()` ignoring a `NoSuchElementException` and `StaleElementReferenceException`

Reply:
Good point, let me re-write it.